### PR TITLE
feat: upgrade to remix v1.18.0; update dependencies

### PR DESCRIPTION
**Pull request by onemen**
done

### CODE REVIEW

**Comment by onemen** [comment]
This is not a new issue, I have seen it before you installed of `remark-emoji`.
`remark-emoji` exist in my `node_modules` but remix does not find it
```shell
[build:remix ]  warn  could not resolve "remark-emoji"
[build:remix ] ┃ You imported "remark-emoji" in utils/compile-mdx.server.ts,
[build:remix ] ┃ but that package is not in your `node_modules`.
[build:remix ] ┃ Did you forget to install it?
[build:remix ]
```

do you see the same?

File: package.json [hunk 1]
```diff
@@ -22,9 +22,9 @@
     "packages/example/examples/*"
   ],
   "devDependencies": {
-    "@playwright/test": "^1.34.3",
-    "@remix-run/eslint-config": "^1.17.0",
-    "dotenv": "^16.1.4",
+    "@playwright/test": "^1.35.1",
+    "@remix-run\/eslint-config": "^1.18.0",
```
Comment (kentcdodds) [inline package.json:26]:
```suggestion
    "@remix-run/eslint-config": "^1.18.0",
```

Pretty sure this was a mistake.

File: packages/example/exercises/01.nested-routing/01.problem.outlet/package.json [hunk 2]
```diff
@@ -21,11 +21,11 @@
...
-    "@remix-run/express": "^1.17.0",
-    "@remix-run/node": "^1.17.0",
-    "@remix-run/react": "^1.17.0",
-    "@remix-run/serve": "^1.17.0",
-    "@remix-run/server-runtime": "^1.17.0",
+    "@remix-run\/express": "^1.18.0",
+    "@remix-run\/node": "^1.18.0",
+    "@remix-run\/react": "^1.18.0",
+    "@remix-run\/serve": "^1.18.0",
+    "@remix-run\/server-runtime": "^1.18.0",
```
Comment (kentcdodds) [inline packages/example/exercises/01.nested-routing/01.problem.outlet/package.json:28]:
```suggestion
    "@remix-run/express": "^1.18.0",
    "@remix-run/node": "^1.18.0",
    "@remix-run/react": "^1.18.0",
    "@remix-run/serve": "^1.18.0",
    "@remix-run/server-runtime": "^1.18.0",
```

Pretty sure this is a mistake as well.

File: packages/example/exercises/01.nested-routing/01.problem.outlet/package.json [hunk 3]
```diff
@@ -45,15 +45,15 @@
     "unist-util-visit": "^4.1.2"
   },
   "devDependencies": {
-    "@remix-run/dev": "^1.17.0",
-    "@remix-run/eslint-config": "^1.17.0",
-    "@types/eslint": "^8.40.0",
+    "@remix-run\/dev": "^1.18.0",
+    "@remix-run\/eslint-config": "^1.18.0",
```
Comment (kentcdodds) [inline packages/example/exercises/01.nested-routing/01.problem.outlet/package.json:49]:
```suggestion
    "@remix-run/dev": "^1.18.0",
    "@remix-run/eslint-config": "^1.18.0",
```

And this

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Could you look at the package.json issue?

I'm not sure what's up with that package. We don't use it in this project.

**Comment by kentcdodds** [comment]
Whoops, we do use it in this project. But I haven't had any issues with it.

File: package.json (hunk 1 above)
Comment (onemen) [reply package.json]: oops

**Comment by onemen** [comment]
> Whoops, we do use it in this project. But I haven't had any issues with it.

do you see the warning in your console ?

**Comment by kentcdodds** [comment]
I hadn't noticed that warning, but yes, it's in there. Things seem to work ok though.

**Review by kentcdodds — APPROVED** [review]
Thank you!

**Pull request merged**